meetings/2022-11/nov-29.md

```diff
@@ -0,0 +1,1110 @@
+
+
+BSH: Just briefly one of the in favor of static methods is if is a static method, then the argument can be any iterable, not just an array, which seems like a more generally useful structure so that might be feature creep for this proposal. I realize, but it could be with this
+
+JRL: Because iterables could be infinite, they’re a bit different. Here you're only taking a fixed amount, the behavior will collates all keys. I would say like if you have a span of 1, 2,1,1 those all 1s get grouped into the same key. But in an iterable,  if you have 1, 2, 1,1 you have 3 separate groups, [1], [2], [1, 1]. The behavior for iterator groupBy has to return itself an iterator which does not have all keys  collated together, only contiguous runs collated.
+
+BSH: Okay, thanks.
+
+USA: All right. That concludes this topic. I would like to reaffirm that this proposal stays stays at stage 3.
+
+ \
+JRL: Yeah. so I would like to remain at stage 3. I'm going to come back next meeting with a formalized spec for static methods and we can agree on that point if we want to approach, if we want to switch to the static method approach, we could either demote to stage two and let it bake with the community at state stage 3.
+
+
+### Conclusion/Decision
+
+
+
+* Proposal stays at stage 3
+* Tentatively going for static methods on Object/Map
+
+
+## Should we set a lower bound on the resolution of timers
+
+Presenter: Shu-yu Guo (SYG)
+
+- [issue](https://github.com/tc39/ecma262/issues/2914)
+
+SYG: OK. This is a pedantic thing that I don't think has any. Really real. world impact. One way or another. currently, but it'd be good to to get clarified in the spec. so Richard Gibson here filed an issue a while back currently. What time? weight is one of those things where we hand wave some stuff about what time actually is and the way that it does the blocking. So sorry called hummocks, not way is a blocking call that suspends the thread of execution until it is notified. This cannot be called on the main thread on the we. because it blocks execution. And the way it does this, it can have a timeout. and if it has a has a timeout, it basically, it goes through this AO called `suspendAgent`, and it just says like, wait for that time. The issue is that we don't really put any normative guidance around what exactly that ‘wait for that time’ means. Does it mean a compliant implementation must wait exactly that time Does it mean it can? Wait, at a coarser granularity of time? But we don't say that. So I think we should say something about that. specifically, Richard Gibson brings up the point of if you can't have, fractional seconds, which is interpreted as factual milliseconds and You observe different things on different implementations. As he shows in this. example here. So, So, What? should we do?
```

Comment:
```suggestion
SYG: OK. This is a pedantic thing that I don't think has any real world impact one way or another currently, but it'd be good to to get clarified in the spec. So RGN here filed an issue a while back currently. `Atomics.wait` is one of those things where we hand wave some stuff about what time actually is and the way that it does the blocking. It is a blocking call that suspends the thread of execution until it is notified. This cannot be called on the main thread because it blocks execution. And the way it does this, it can have a timeout that goes to this AO called SuspendAgent, and it just says like, wait for that time. The issue is that we don't really put any normative guidance around what exactly that ‘wait for that time’ means. Does it mean a compliant implementation must wait exactly that time? Does it mean it can wait at a coarser granularity of time? We don't say. So I think we should say something about that. Specifically, RGN brings up the point of fractional milliseconds, and you observe different things on different implementations. As he shows in this. example here. So, what should we do?
```